x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/parser/ExpressionBuilder.java

```diff
@@ -118,17 +124,17 @@ public Expression visitComparison(ComparisonContext ctx) {
 
         switch (op.getSymbol().getType()) {
             case EqlBaseParser.EQ:
-                return new Equals(source, left, right);
+                return new Equals(source, left, right, params.zoneId());
```

Comment:
Similar to source, just extract the params.zoneId into a variable an use that instead - less verbose and makes potential refactoring easier in the future.